### PR TITLE
AArch64: Introduce a pseudo real register for killing vector registers

### PR DESCRIPTION
**Pull request by Akira1Saitoh**
Introduce `spillAllVectorRegs` helper function to OMRMachine class.
Introduce `KillVectorRegs` pseudo real register.
Update the register dependency class to kill all vector registers by  calling `spillAllVectorRegs` helper function
when `KillVectorRegs` is in the dependencies.

Signed-off-by: Akira Saitoh <saiaki@jp.ibm.com>

### CODE REVIEW

File: compiler/aarch64/codegen/OMRRegisterDependency.cpp [hunk 1]
```diff
@@ -419,6 +419,18 @@ void OMR::ARM64::RegisterDependencyConditions::stopUsingDepRegs(TR::CodeGenerato
...
    }
 
+
+/**
+ * @brief Answers if the register dependency is KillVectorRegs
+ *
+ * @param[in] dep : register dependency
+ * @return true if the register dependency is KillVectorRegs
+ */
+static bool isKillVectorRegs(TR::RegisterDependency *dep)
```
Comment (knn-k) [inline compiler/aarch64/codegen/OMRRegisterDependency.cpp:429]: I think `killsVectorRegs()` would be a better name for this function.

File: compiler/aarch64/codegen/OMRMachine.cpp [hunk 2]
```diff
@@ -314,12 +308,23 @@ TR::RealRegister *OMR::ARM64::Machine::freeBestRegister(TR::Instruction *current
...
+   self()->spillRegister(currentInstruction, candidates[0]);
+   return best;
+   }
+
+void OMR::ARM64::Machine::spillRegister(TR::Instruction *currentInstruction, TR::Register *virtReg)
+   {
+   TR::Register *registerToSpill = virtReg;
+   TR::CodeGenerator *cg = self()->cg();
+   TR::Compilation *comp = cg->comp();
+   TR_Debug *debugObj = self()->cg()->getDebug();
```
Comment (knn-k) [inline compiler/aarch64/codegen/OMRMachine.cpp:320]: Use `cg` declared above.

File: compiler/aarch64/codegen/OMRMachine.cpp [hunk 3]
```diff
@@ -314,12 +308,23 @@ TR::RealRegister *OMR::ARM64::Machine::freeBestRegister(TR::Instruction *current
...
+   TR_RegisterKinds rk = virtReg->getKind();
+   TR::Node * currentNode = currentInstruction->getNode();
```
Comment (knn-k) [inline compiler/aarch64/codegen/OMRMachine.cpp:322]: Please remove the space between `*` and `currentNode`.

File: compiler/aarch64/codegen/OMRMachine.cpp [hunk 4]
```diff
@@ -899,6 +905,26 @@ void OMR::ARM64::Machine::coerceRegisterAssignment(TR::Instruction *currentInstr
...
+void OMR::ARM64::Machine::spillAllVectorRegisters(TR::Instruction  *currentInstruction)
+   {
+   int32_t first = TR::RealRegister::FirstFPR;
+   int32_t last  = TR::RealRegister::LastFPR;
+   TR::Node *node = currentInstruction->getNode();
+
+   for (int32_t i = first; i <= last; i++)
+      {
+      TR::Register * virtReg = NULL;
+      TR::RealRegister * realReg = self()->getRealRegister(static_cast<TR::RealRegister::RegNum>(i));
```
Comment (knn-k) [inline compiler/aarch64/codegen/OMRMachine.cpp:917]: Please remove the space between `*` and the variable names.

File: compiler/aarch64/codegen/OMRRegisterDependency.cpp (hunk 1 above)
Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRRegisterDependency.cpp]: Changed the function name to `killVectorRegs`.

File: compiler/aarch64/codegen/OMRMachine.cpp (hunk 2 above)
Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRMachine.cpp]: Fixed it.

File: compiler/aarch64/codegen/OMRMachine.cpp (hunk 3 above)
Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRMachine.cpp]: Removed the space.

File: compiler/aarch64/codegen/OMRMachine.cpp (hunk 4 above)
Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRMachine.cpp]: Removed the spaces.

**Comment by Akira1Saitoh** [comment]
I also updated `getRegisterName()` in ARM64Debug.cpp to return the name for `KillVectorRegs` properly.

File: compiler/aarch64/codegen/OMRRegisterDependency.cpp (hunk 1 above)
Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRRegisterDependency.cpp]: Changed the function name again to `killsVectorRegs`.

**Comment by knn-k** [comment]
Jenkins build aarch64

**Review by knn-k — APPROVED** [review]

**Pull request merged**